src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -1059,6 +1096,13 @@ trait Implicits {
        *   - find the most likely one
        *   - if it matches, forget about all others it improves upon
        */
+
+      // the pt for views can have embedded unification type variables, BoundedWildcardTypes or
+      // Nothings which can't be solved for. Rather than attempt to patch things up later we
+      // just skip those cases altogether.
+      lazy val wildPtIsInstantiable =
+        !wildPt.exists { case _: BoundedWildcardType | _: TypeVar => true ; case tp if typeIsNothing(tp) => true; case _ => false }
```

Comment:
I'm not convinced we should look for `Nothing` here. Also, we could remove the double negation (it's only used as `!wildPtIsInstantiable`) and call this `wildPtNotFullyDefined`. That then leads to another simplification: the RHS could be just `!isFullyDefined(wildPt)`.

Reply:
Point taken on the double negation ... I'll fix.

We really do need the check for `Nothing` though: see `test/files/pos/prune-poly-infer-nothing.scala` without the check for `Nothing` the correctly inferred `Nothing` for `O` would be retracted resulting in a false negative. That test was minimized from a failure in http4s in the 2.12 community build.

Aside from that, `isFullyDefined` is too restrictive ... it rules out cases which should be allowed (eg. any case where the `pt` contains a vanilla wildcard).